lxd/seccomp/seccomp.go

```diff
@@ -1516,6 +1531,78 @@ func (s *Server) HandleSetxattrSyscall(c Instance, siov *Iovec) int {
 	return 0
 }
 
+// SchedSetschedulerArgs arguments for setxattr.
+type SchedSetschedulerArgs struct {
+	nsuid   int64
+	nsgid   int64
+	nsfsuid int64
+	nsfsgid int64
+	size    int
+	pid     int
+
+	target C.int
+	policy C.int
+}
+
+// HandleSetxattrSyscall handles setxattr syscalls.
+func (s *Server) HandleSchedSetschedulerSyscall(c Instance, siov *Iovec) int {
+	ctx := log.Ctx{"container": c.Name(),
+		"project":               c.Project(),
+		"syscall_number":        siov.req.data.nr,
+		"audit_architecture":    siov.req.data.arch,
+		"seccomp_notify_id":     siov.req.id,
+		"seccomp_notify_flags":  siov.req.flags,
+		"seccomp_notify_pid":    siov.req.pid,
+		"seccomp_notify_fd":     siov.notifyFd,
+		"seccomp_notify_mem_fd": siov.memFd,
+	}
+
+	defer logger.Debug("Handling sched_setscheduler syscall", ctx)
+
+	args := SchedSetschedulerArgs{}
+	args.pid = int(siov.req.pid)
+
+	pidFdNr, pidFd := MakePidFd(args.pid, s.s)
```

Comment:
The `pid` argument to `sched_setscheduler()` can be a non-threadgroup leader, i.e., a thread id and `pidfd_open()` doesn't work on single threads yet, unfortunately.

Reply:
@brauner that is a good point and actually quite important in our case. We have Android calling sched_setscheduler for specific threads so `siov.req.pid` will end up being a tid and the `MakePidFd` call will failed. The only way I see right now to fix this is by parsing `/proc/$(siov.req.pid)/status` and check the `Tgid` field to get the actual parent pid we can then use to join the right namespace.

Reply:
cc @stgraber